general-superstaq/general_superstaq/superstaq_exceptions.py

```diff
@@ -8,23 +8,33 @@
 class SuperstaqException(Exception):
     """An exception for errors coming from Superstaq's API."""
 
-    def __init__(self, message: str, status_code: Optional[int] = None):
+    def __init__(self, message: str, status_code: Optional[int] = None) -> None:
         """Initializes the `SupertaqException` class.
 
         Args:
             status_code: An HTTP status code, if coming from an HTTP response with a failing
                 status.
             message: A message corresponding to the HTTP response status code. Defaults to None.
         """
-        super().__init__(f"Status code: {status_code}, Message: '{message}'")
         self.status_code = status_code
         self.message = message
 
+        if status_code is None:
+            err_msg = f"{message}."
+        else:
+            status_msg = (
+                "400, non-retriable error making request to Superstaq API"
+                if (status_code == 400)
+                else str(status_code)
+            )
+            err_msg = f"{message}. (Status code: {status_msg})"
```

Comment:
sounds good, i decided to include the punctuation because i noticed some error messages did not include it, but i guess it's a different issue to make sure we use proper punctuation everywhere we raise this. also i didn't think of the question mark case, good catch.